lips/lip-20.md

```diff
@@ -0,0 +1,595 @@
+In addition to the metadata that each currency must possess, each currency has a number of
+[events](TODO: point to event doc) that are associated with it, and that are emitted when
+the metadata for a currency is changed. These events are as follows:
+
+| Name                          | Type                                                                                                                                                     | Description                                                                                                 |
+| ----------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------- |
+| mint_events                   | EventHandle\<[MintEvent](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#struct-mintevent)\>                             | Event emitted whenever SCSs are created on-chain                                                            |
+| burn_events                   | EventHandle\<[BurnEvent](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#struct-burnevent)\>                             | Event emitted whenever SCSs are removed on-chain                                                            |
+| preburn_events                | EventHandle\<[PreburnEvent](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#struct-preburnevent)\>                       | Event emitted whenever SCSs are moved to a preburn area                                                     |
+| cancel_burn_events            | EventHandle\<[CancelBurnEvent](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#struct-cancelburnevent)\>                 | Event emitted whenever SCSs in a preburn area are moved back to circulation                                 |
+| exchange_rate_update_events   | EventHandle\<[ExchangeRateUpdateEvent](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#struct-exchangerateupdateevent)\> | Event emitted whenever the `to_lbr_exchange_rate` metadata field is updated (for all registered currencies) |
+
+---
+# <a name="operational_things">Operational Restrictions</a>
+---
+
+When deciding the on-chain currency code for a currency,
+the module, as well as the type name chosen for the currency must match the on-chain currency
+code chosen for the currency. e.g., if you want to use currency code `"ABC"` for a
+currency, then the fully qualified Move type that is used for that currency on-chain
+_must_ be `0x1::ABC::ABC`.
+
+Because of this, and in addition:
+* An on-chain currency code must be a valid Move identifier; and
+* It can consist solely of alphanumeric (TODO: [uppercase](https://github.com/libra/libra/issues/6082)) characters.
```

Comment:
Are we planning to resolve this TODO when we do the coin rename?

Reply:
yep! Although I think I'll remove the TODO since it isn't really adding anything IMO